File: springproy/demo/src/main/java/sistema/reservas/interurbanas/Reserva.java
package sistema.reservas.interurbanas;

import java.time.LocalDateTime;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;

@Entity
@Table(name = "reservas")
public class Reserva {
	@Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
	
	@ManyToOne
    private Usuario usuario;
	
	@ManyToOne
    private Ruta ruta;
	
	private LocalDateTime fechaHora;
    private int asiento;

}

File: front/app/src/main/java/com/example/proyectofinaltransporte/network/Reserva.kt
package com.example.proyectofinaltransporte.network

import kotlinx.serialization.Contextual
import kotlinx.serialization.Serializable
import java.time.LocalDate



@Serializable
class Reserva (
    val id: Long? = null,
    val fechaHora: String,
    val asiento: Int,
    val ruta: Ruta,
    val vehiculo: Vehiculo,
)